### DISCUSSION TITLE
NPM package downloads

### DISCUSSION THREAD

**Discussion by EmmaNingMS** [community/community#145003] [npm] [answered]
Curious how NPM tracks downloads in the dependency tree. For example, if NPM package A depends on package B, will the downloads of package A be counted as downloads for package B?

**Comment by monishcm** [comment, marked as answer]
This [link](https://blog.npmjs.org/post/92574016600/numeric-precision-matters-how-npm-download-counts-work.html) (though archived) can give you a perspective. 

In summary, when a package is downloaded directly from the npm registry (be it a dependent or a direct fetch), its download count increases. However, various factors, such as local caching, mirrors, or proxies, can bypass direct interactions with the registry. As a result, these scenarios do not contribute to the package’s download count.
